mls-rs/src/group/proposal_filter/filtering.rs

```diff
@@ -460,34 +517,58 @@ pub(crate) fn proposer_can_propose(
     source: &ProposalSource,
 ) -> Result<(), MlsError> {
     let can_propose = match (proposer, source) {
-        (Sender::Member(_), ProposalSource::ByValue | ProposalSource::Local) => matches!(
-            proposal_type,
-            ProposalType::ADD
-                | ProposalType::REMOVE
-                | ProposalType::PSK
-                | ProposalType::RE_INIT
-                | ProposalType::GROUP_CONTEXT_EXTENSIONS
-        ),
-        (Sender::Member(_), ProposalSource::ByReference(_)) => matches!(
-            proposal_type,
-            ProposalType::ADD
-                | ProposalType::UPDATE
-                | ProposalType::REMOVE
-                | ProposalType::PSK
-                | ProposalType::RE_INIT
-                | ProposalType::GROUP_CONTEXT_EXTENSIONS
-        ),
+        (Sender::Member(_), ProposalSource::ByValue | ProposalSource::Local) => {
+            let can_propose = matches!(
+                proposal_type,
+                ProposalType::ADD
+                    | ProposalType::REMOVE
+                    | ProposalType::PSK
+                    | ProposalType::RE_INIT
+                    | ProposalType::GROUP_CONTEXT_EXTENSIONS
+            );
+            #[cfg(all(
+                feature = "by_ref_proposal",
+                feature = "custom_proposal",
+                feature = "self_remove_proposal"
+            ))]
+            let can_propose = can_propose || matches!(proposal_type, ProposalType::SELF_REMOVE);
+            can_propose
```

Comment:
I don't think self remove can be by value - that would cause "committer can't self remove"? But it could be local I guess

Reply:
Ah good point! Youre totally right. I've split up the `Local` and `ByValue` cases, and removed the self-remove proposal from the `ByValue` case